lib/index.js

```diff
@@ -92,12 +94,54 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
         });
     }));
 
+  app.get(
+    adminPath,
+    ensureAuthenticated,
+    intParser(),
+    validate({query: 'account.pagination'}),
+    asyncHandler(async (req, res) => {
+      const {email, cursor = null, limit = 10} = req.query;
+      const {actor} = req.user || {};
+      const query = {"account.email": email};
+      if(cursor) {
+        query["account.created"] = {$gte: cursor};
+      }
+      const options = {limit, sort: {"account.created": -1}};
+      const records = await brAccount
+        .getAll({actor, query, options});
+      res.status(200).json(records);
+    }));
+
   app.patch(
     accountPath,
     ensureAuthenticated,
-    asyncHandler(async (req, res, next) => {
-      // TODO: call `brAccount.update`
-      next();
+    intParser(),
+    validate('account.update'),
+    asyncHandler(async (req, res) => {
+      /**
+       * @function accountUpdateHandler
+       * @param {Request} req
+       * @param {Response} res
+       * @param {Function} next
+       * @description patch requests to /:account result in an updated account
+       * patches need to be in the:
+       * [json patch]{@link https://www.npmjs.com/package/fast-json-patch}
+       * format. The body needs to contains the id, patch, and sequence keys
+       */
+      try {
+        const {actor} = req.user || {};
+        const {account: id} = req.params;
+        const updater = {
+          actor,
+          id,
+          ...req.body
+        };
+        await brAccount.update(updater);
+        res.status(200).send();
+      } catch(e) {
+        console.error(e);
```

Comment:
inside asyncHandler inside bedrock, you just want to throw the BedrockError, which ultimately calls the next() handler which at the end (end bedrock) does the right thing with errors.  That's why BedrockError details include `httpStatusCode`.  Also, never `console.log` anything in production code.  as it happens, bedrock-express logs errors using the bedrock logging functionality.  If you ever need to setup a logger, here's an example: https://github.com/digitalbazaar/bedrock-ledger-node/blob/master/lib/logger.js

Reply:
cool will do

Reply:
the error type I remember seeing a list of them, but don't have the link on me.

Reply:
What I'm suggesting here is that you just let brAccount.update throw if it needs to throw (no try/catch) and the error it throws should have the proper BedrockError with a status code, which needs investigation.  For future reference, here's the error types: https://developer.mozilla.org/en-US/docs/Web/API/DOMException

Reply:
ok cool will do that now.